### ISSUE TITLE
Allow to zoom to any layer or object3D from the viewer

### ISSUE THREAD

**Issue by jailln** [iTowns/itowns#1906]
I think that an helpful API enhancement would be to add a `zoomTo` method to `View` that could either take a layer id or an object3D as parameter. 

To zoom to a layer, we could use the `boundingBox` attribute of layers (see #1903 ) and the CameraUtils methods (see #1904).
To zoom to a 3D object, we could also use CameraUtils methods after computing the Object3D bounding box with something like this:
```
var box = new THREE.Box3();
box.setFromObject(object3d);
```

MapBox and Cesium have a similar API for this usecase (`map.zoomTo()` or `viewer.zoomTo()` respectively) and it's actually quite useful.

**Comment by mgermerie** [comment]
Hi, thanks for the suggestion !
I agree that it could be useful to have a method in `View` allowing to zoom on a layer data or an object3d. 

There is a method in `CameraUtils` called [transformCameraToLookAtTarget](https://github.com/iTowns/itowns/blob/91fd9ec49686148698a5f7577644e1310ff41168/src/Utils/CameraUtils.js#L397) which can zoom to a given `Extent`. It is used in [this example](http://www.itowns-project.org/itowns/examples/index.html#source_file_geojson_raster) to zoom to a `ColorLayer` data extent. 
I don't know if the `Source` of any `Layer` has an extent defined. If that is the case, we could just zoom on this extent for the layers, without needing to compute `boundingBox` for each Layer.

In any case, it would indeed be nice to have it available in the `View`.

I shall investigate a bit more on this and let you know of my findings.

**Comment by jailln** [comment]
I agree with you that we don't need a `boundingBox` for each layer to implement a `zoomTo` method. However, I still think that each layer should have a `boundingBox` attribute because it is not really costly to compute nor to store and it is commonly used in many processes that itowns' users might implement and also commonly available in other geospatial visualization frameworks, so it's not only related to this issue for me (that's why I opened two separated issues).

> There is a method in `CameraUtils` called [transformCameraToLookAtTarget](https://github.com/iTowns/itowns/blob/91fd9ec49686148698a5f7577644e1310ff41168/src/Utils/CameraUtils.js#L397) which can zoom to a given `Extent`. It is used in [this example](http://www.itowns-project.org/itowns/examples/index.html#source_file_geojson_raster) to zoom to a `ColorLayer` data extent. I don't know if the `Source` of any `Layer` has an extent defined. If that is the case, we could just zoom on this extent for the layers, without needing to compute `boundingBox` for each Layer.

Yes actually there is and I used the same logic to zoom to a bounding box in #1904. However, I think we still need a method to zoom to a bounding box for several reasons:
* It takes into account the Z dimension to compute the range of the camera while an extent does not
* Zooming to a 3D bounding box is more flexible in the sense it doesn't suppose that the objects it is zooming to are on the ground and it can be easily used for threejs 3D objects.
* [transformCameraToLookAtTarget](https://github.com/iTowns/itowns/blob/91fd9ec49686148698a5f7577644e1310ff41168/src/Utils/CameraUtils.js#L429) currently sets the tilt to 90 by default when zooming to an extent, but this could (should?) be made configurable.
* Some layers (e.g. 3D Tiles layers) don't have an extent defined (we could compute it from the bounding box but that would mean computing the bounding box anyway).

**Comment by mgermerie** [comment]
We could implement the possibility to pass bounding box as third parameter of [transformCameraToLookAtTarget](https://github.com/iTowns/itowns/blob/91fd9ec49686148698a5f7577644e1310ff41168/src/Utils/CameraUtils.js#L397).
